sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -405,23 +427,28 @@ public virtual Response DeleteConfigurationSetting(ConfigurationSetting setting,
 
         private async Task<Response> DeleteConfigurationSettingAsync(string key, string label, MatchConditions requestOptions, CancellationToken cancellationToken = default)
         {
-            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ConfigurationClient)}.{nameof(DeleteConfigurationSetting)}");
+            using DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(ConfigurationClient)}.{nameof(DeleteConfigurationSetting)}");
             scope.AddAttribute("key", key);
             scope.Start();
 
             try
             {
-                using Request request = CreateDeleteRequest(key, label, requestOptions);
-                Response response = await _pipeline.SendRequestAsync(request, cancellationToken).ConfigureAwait(false);
+                var eTag = requestOptions?.IfMatch;
+                if (requestOptions != null && requestOptions.IfMatch.HasValue)
+                {
+                     eTag = new ETag($"\"{requestOptions?.IfMatch.Value.ToString()}\"");
```

Comment:
Looking at [ConditionalRequestOptionsExtensions.ApplyHeaders](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/ConditionalRequestOptionsExtensions.cs#L14-L15), we don't need `$"\"{requestOptions?.IfMatch.Value.ToString()}\""`  when `options.IfMatch.Value` is equals to `ETag.All`. Can we apply [this](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/ConditionalRequestOptionsExtensions.cs#L14-L15) similar condition here?

Also, could you please create an issue in [c# generator](https://github.com/Azure/autorest.csharp/issues/new/choose) about [this](https://github.com/DivineNwaokocha/azure-sdk-for-net/blob/43a9c2ff42fa0b84b01a30831b15d8ba8fcc2a19/sdk/appconfiguration/Azure.Data.AppConfiguration/src/Generated/ConfigurationClient.cs#L2113) should call [ConditionalRequestOptionsExtensions.ApplyHeaders](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/ConditionalRequestOptionsExtensions.cs#L10) method when applying `MatchConditions` header, so we don't have to add this here?